site/documentation/content/api/command-and-control-kafka/index.md

```diff
@@ -0,0 +1,117 @@
+The Business Application can consume the corresponding command response from the `hono.command_response.${tenant_id}` topic.
+
+**Preconditions**
+
+1. Either the topics `hono.command.${tenant_id}` and `hono.command_response.${tenant_id}` exist, or the broker is configured to automatically create topics on demand.
+1. The *Business Application* is authorized to write to the `hono.command.${tenant_id}` topic and read from the `hono.command_response.${tenant_id}` topic.
+1. The *Business Application* subscribes to the `hono.command_response.${tenant_id}` topic with a Kafka consumer.
+
+**Message Flow**
+
+1. The *Business Application* produces a command message to the *Kafka Cluster* on the topic `hono.command.${tenant_id}`.
+1. Hono consumes the message from the *Kafka Cluster* and forwards it to the device, provided that device is connected and accepting commands.
+1. The device sends a command response message. Hono writes that message into the `hono.command_response.${tenant_id}` topic on the *Kafka Cluster*.
+1. The *Business Application* consumes the command response message from the `hono.command_response.${tenant_id}` topic.
+
+**Command Message Format**
+
+The key of the message MUST be the ID of the device that the command is targeted at.
+
+Metadata MUST be set as Kafka headers on a message.
+The following table provides an overview of the headers the *Business Application* needs to set on a request/response command message.
+
+| Name               | Mandatory | Type      | Description |
+| :----------------- | :-------: | :-------- | :---------- |
+| *device_id*        | yes       | *string*  | The identifier of the device that the command is targeted at. |
+| *subject*          | yes       | *string*  | The name of the command to be executed by the device. |
+| *expect-response*  | yes       | *boolean* | MUST be set with a value of `true` for a command for which a response from the device is expected. |
+| *content-type*     | no        | *string*  | If present, MUST contain a *Media Type* as defined by [RFC 2046](https://tools.ietf.org/html/rfc2046) which describes the semantics and format of the command's input data contained in the message payload. However, not all protocol adapters will support this property as not all transport protocols provide means to convey this information, e.g. MQTT 3.1.1 has no notion of message headers. |
+| *correlation-id*   | no        | *string*  | MAY contain an ID used to correlate a response message to the original request. If set, it is used as the *correlation-id* header in the response, otherwise the value of the *message-id* header is used. Either this or the *message-id* header MUST be set. |
+| *message-id*       | no        | *string*  | MAY contain an identifier that uniquely identifies the message at the sender side. Either this or the *correlation-id* header MUST be set. |
```

Comment:
do we need this? why don't we simply require the use of *correlation-id*?

Reply:
Right, for AMQP it might make sense to have the `message-id` fallback since that is a system-property there, but not so for Kafka, since it's another custom header there. I'll remove it.